views/sidecolumn.html

```diff
@@ -1,36 +1,91 @@
-                    hx-trigger="click"
-                    hx-select-oob="#sidecolumn"
-            >
-            <label for="{{ .Name }}" class="text-gray-700">{{ .Name }}</label>
+<aside id="sidecolumn" class="mx-6 mt-6 w-64 rounded-md bg-white">
+    {{ range $index, $filter := .Filters }}
+    <div id="accordion-{{ $index }}" class="border-b border-gray-200">
+      <!-- Accordion Header -->
+      <div id="accordion-heading-{{ $index }}" class="p-4 cursor-pointer bg-gray-100 hover:bg-gray-200">
+        <button
+        id="accordion-button-{{ $index }}"
+        aria-expanded="false"
+        aria-controls="accordion-body-{{ $index }}"
+        class="w-full"
+        onclick="toggleAccordion('{{ $index }}')">
+        <div class="flex items-center justify-between w-full spaxe-x-10">
+          <span class="text-black font-bold">{{ $filter.Category }}</span>
+          <div class="svg-wrapper w-4 h-4 flex-shrink-0">
+            <svg 
+              class="w-full h-full transform transition-transform duration-300" 
+              xmlns="http://www.w3.org/2000/svg" 
+              fill="none" 
+              viewBox="0 0 14 8">
+              <path 
+                stroke="currentColor" 
+                stroke-linecap="round" 
+                stroke-linejoin="round" 
+                stroke-width="1" 
+                d="M1 1l6 6 6-6"></path>
+            </svg>
+          </div>
         </div>
-        {{end}}
-    </div>
-</div>
-{{end}}
+      </button>
+      </div>
 
-{{ define "sidecolumn-nohtmx" }}
-<div id="sidecolumn" class="mt-12 ml-12">
-    <p>{{ .Count }} Results</p>
-    <div class="flex items-center space-x-2">
-        <div class="flex items-center space-x-2">
-            <input type="checkbox">
-            <label class="text-gray-700">Filter</label>
+      <!-- Accordion Body -->
+      <div
+        id="accordion-body-{{ $index }}"
+        role="region"
+        aria-labelledby="accordion-button-{{ $index }}"
+        class="hidden-filters p-4 bg-white">
+        <div class="space-y-2">
+          <!-- Dynamic Content with Checkboxes -->
+          {{ range $optionIndex, $option := $filter.Options }}
+          <div class="flex items-center justify-between">
+            <label for="filter-{{ $index }}-{{ $optionIndex }}" class="flex items-center space-x-2 cursor-pointer">
+              <input
+                id="filter-{{ $index }}-{{ $optionIndex }}"
+                type="checkbox"
+                class="form-checkbox h-4 w-4 text-blue-600">
+              <span>{{ $option.Label }}</span>
+            </label>
+            <span>{{ $option.Count }}</span>
+          </div>
+          {{ end }}
         </div>
+      </div>
     </div>
-</div>
-{{end}}
+    {{ end }}
+</aside>
+
+<style>
+  .hidden-filters {
+    display: none !important;
+  }
+  .svg-wrapper {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 16px;
+    height: 16px;
+  }
+</style>
```

Comment:
instead of using a style tag, can we just directly toggle the tailwind classes like this so that we don't need to define a style classes?

Reply:
It may be even possible to get rid of the javascript by defining something like this in tailwind input.css. 

```@layer utilities {
  .hidden-default {
    @apply hidden opacity-0;
  }

  .hidden-fade {
    @apply hidden opacity-0 transition-opacity duration-300;
  }

  .hidden-slide {
    @apply hidden transform -translate-y-4 transition-transform duration-300;
  }
}```

Then your html would look like something like:

```
<div id="accordion-1" class="hidden-default">
  <p>This is a hidden accordion section.</p>
</div>

<button onclick="toggleVisibility('accordion-1', 'hidden-fade')">
  Toggle Accordion with Fade
</button>

<button onclick="toggleVisibility('accordion-1', 'hidden-slide')">
  Toggle Accordion with Slide
</button>

```

Reply:
You could use the classes in html like this

```html
<div id="accordion-1" class="hidden-default">
  <p>This is a hidden accordion section.</p>
</div>

<button onclick="toggleVisibility('accordion-1', 'hidden-fade')">
  Toggle Accordion with Fade
</button>

<button onclick="toggleVisibility('accordion-1', 'hidden-slide')">
  Toggle Accordion with Slide
</button>